src/super_gradients/training/models/pose_estimation_models/yolo_nas_pose/yolo_nas_pose_variants.py

```diff
@@ -0,0 +1,324 @@
+    def infer_total_number_of_predictions(self, inputs: Any) -> int:
+        """
+
+        :param inputs: YoloNASPose model outputs
+        :return:
+        """
+        if torch.jit.is_tracing():
+            pred_bboxes_xyxy, pred_bboxes_conf, pred_pose_coords, pred_pose_scores = inputs
+        else:
+            pred_bboxes_xyxy, pred_bboxes_conf, pred_pose_coords, pred_pose_scores = inputs[0]
+
+        return pred_bboxes_xyxy.size(1)
+
+    def get_num_pre_nms_predictions(self) -> int:
+        return self.num_pre_nms_predictions
+
+    def forward(self, inputs: Tuple[Tuple[Tensor, Tensor], Tuple[Tensor, ...]]):
+        """
+        Decode YoloNASPose model outputs into bounding boxes, confidence scores and pose coordinates and scores
+
+        :param inputs: YoloNASPose model outputs
+        :return: Tuple of (pred_bboxes, pred_scores, pred_joints)
+        - pred_bboxes: [Batch, num_pre_nms_predictions, 4] Bounding of associated with pose in XYXY format
+        - pred_scores: [Batch, num_pre_nms_predictions, 1] Confidence scores [0..1] for entire pose
+        - pred_joints: [Batch, num_pre_nms_predictions, Num Joints, 3] Joints in (x,y,confidence) format
+        """
+        if torch.jit.is_tracing():
+            pred_bboxes_xyxy, pred_bboxes_conf, pred_pose_coords, pred_pose_scores = inputs
+        else:
+            pred_bboxes_xyxy, pred_bboxes_conf, pred_pose_coords, pred_pose_scores = inputs[0]
+
+        nms_top_k = self.num_pre_nms_predictions
+
+        topk_candidates = torch.topk(pred_bboxes_conf, dim=1, k=nms_top_k, largest=True, sorted=True)
+
+        offsets = nms_top_k * torch.arange(pred_bboxes_conf.size(0), device=pred_bboxes_conf.device)
+        flat_indices = topk_candidates.indices + offsets.reshape(pred_bboxes_conf.size(0), 1)
+        flat_indices = torch.flatten(flat_indices)
+
+        pred_poses_and_scores = torch.cat([pred_pose_coords, pred_pose_scores.unsqueeze(3)], dim=3)
+
+        output_pred_bboxes = pred_bboxes_xyxy.reshape(-1, pred_bboxes_xyxy.size(2))[flat_indices, :].reshape(
+            pred_bboxes_xyxy.size(0), nms_top_k, pred_bboxes_xyxy.size(2)
+        )
+        output_pred_scores = pred_bboxes_conf.reshape(-1, pred_bboxes_conf.size(2))[flat_indices, :].reshape(
+            pred_bboxes_conf.size(0), nms_top_k, pred_bboxes_conf.size(2)
+        )
+        output_pred_joints = pred_poses_and_scores.reshape(-1, pred_poses_and_scores.size(2), 3)[flat_indices, :, :].reshape(
+            pred_poses_and_scores.size(0), nms_top_k, pred_poses_and_scores.size(2), pred_poses_and_scores.size(3)
+        )
+
+        return output_pred_bboxes, output_pred_scores, output_pred_joints
+
+
+class YoloNASPose(CustomizableDetector, ExportablePoseEstimationModel):
+    def __init__(
+        self,
+        backbone: Union[str, dict, HpmStruct, DictConfig],
+        heads: Union[str, dict, HpmStruct, DictConfig],
+        neck: Optional[Union[str, dict, HpmStruct, DictConfig]] = None,
```

Comment:
Not doable. I think we cannot have backbone/heads/neck as instances here, since it is a CustomizableDetector who is actually instantiating stuff. Maybe it is not the most elegant solution as it limits us and forces using configs. But it is how it is.